src/state-selector/create-state-selector.ts

```diff
@@ -0,0 +1,118 @@
+import mitt, { Emitter } from 'mitt';
+import {
+  createReadonlyStateWireGuard,
+  ReadonlyStateWire,
+} from '../state-wire/readonly-state-wire';
+import { createStateWireGuard } from '../state-wire/state-wire';
+import { StateWire } from '../types';
+import { Defined } from '../utils/type-utils';
+
+type DisconnectFunction = () => void;
+type ConnectFunction = () => DisconnectFunction;
+
+export type GetWireValue = <V>(wire: ReadonlyStateWire<V>) => V;
+export type SetWireValue = <V>(wire: StateWire<V>, value: Defined<V>) => void;
+
+export interface ReadOnlySelectorOptions<V> {
+  get: (options: { get: GetWireValue }) => V;
+}
+
+export interface WritableSelectorOptions<V> {
+  get: (options: { get: GetWireValue }) => V;
+  set: (options: { get: GetWireValue; set: SetWireValue }, value: V) => void;
+}
+
+export interface SelectorOptions<V> {
+  get: (options: { get: GetWireValue }) => V;
+  set?: (options: { get: GetWireValue; set: SetWireValue }, value: V) => void;
+}
+
+export function createStateSelector<V>(
+  options: WritableSelectorOptions<V>,
+): [StateWire<V>, ConnectFunction];
+export function createStateSelector<V>(
+  options: ReadOnlySelectorOptions<V>,
+): [ReadonlyStateWire<V>, ConnectFunction];
+export function createStateSelector<V>(
+  options: SelectorOptions<V>,
+): [ReadonlyStateWire<V> | StateWire<V>, ConnectFunction] {
+  const key: string = 'value';
+  const activeWires: {
+    current: Map<ReadonlyStateWire<any>, null | (() => void)>;
+  } = {
+    current: new Map(),
+  };
+  const { get: getOption, set: setOption } = options;
```

Comment:
Maybe `getOption` and `setOption` names are misleading

Reply:
any suggestions?

Reply:
`getValue` and `setValue` ?

Reply:
these names used at line 98